terraform/custom_domains/infrastructure/terraform.tf

```diff
@@ -0,0 +1,22 @@
+terraform {
+
+  required_version = "= 1.3.5"
```

Comment:
Can we use the same version as https://github.com/DFE-Digital/apply-for-qualified-teacher-status/blob/main/terraform/aks/terraform.tf#L2